core/src/test/java/org/elasticsearch/cluster/node/DiscoveryNodesTests.java

```diff
@@ -98,26 +104,98 @@ public void testResolveNodesIds() {
         assertThat(resolvedNodesIds, equalTo(expectedNodesIds));
     }
 
-    private static DiscoveryNodes buildDiscoveryNodes() {
-        int numNodes = randomIntBetween(1, 10);
-        DiscoveryNodes.Builder discoBuilder = DiscoveryNodes.builder();
+    public void testDeltas() {
+        Set<DiscoveryNode> nodesA = new HashSet<>();
+        nodesA.addAll(randomNodes(1 + randomInt(10)));
+        Set<DiscoveryNode> nodesB = new HashSet<>();
+        nodesB.addAll(randomNodes(1 + randomInt(5)));
+        for (DiscoveryNode node : randomSubsetOf(nodesA)) {
+            if (randomBoolean()) {
+                // change an attribute
+                Map<String, String> attrs = new HashMap<>(node.getAttributes());
+                attrs.put("new", "new");
+                node = new DiscoveryNode(node.getName(), node.getId(), node.getAddress(), attrs, node.getRoles(), node.getVersion());
```

Comment:
it actually is - the addition of new attributes means the node had to be restarted and got a new ephemeral id.